<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>教室班牌信息展示系统 V1.0</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_5">
    <!-- 顶部标题栏 -->
    <item>
     <widget class="QLabel" name="titleLabel">
      <property name="text">
       <string>教室班牌信息展示系统</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="styleSheet">
       <string>font-size: 24px; font-weight: bold; color: #2c3e50;</string>
      </property>
     </widget>
    </item>
    <!-- 班级选择+搜索栏 -->
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_1">
      <item>
       <widget class="QLabel" name="label_1">
        <property name="text">
         <string>选择班级：</string>
        </property>
        <property name="styleSheet">
         <string>font-size: 14px;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="classComboBox">
        <property name="minimumSize">
         <size>
          <width>300</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>搜索：</string>
        </property>
        <property name="styleSheet">
         <string>font-size: 14px;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="searchEdit">
        <property name="placeholderText">
         <string>输入班级/教室/院系关键词...</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_1">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="exportBtn">
        <property name="text">
         <string>导出数据</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="noticeManagerBtn">
        <property name="text">
         <string>通知管理</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="settingsBtn">
        <property name="text">
         <string>系统设置</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <!-- 中间内容区（当前课程+课表） -->
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <!-- 左侧：当前/下节课信息 -->
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_1">
        <item>
         <widget class="QFrame" name="currentCourseFrame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>当前课程</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
             <property name="styleSheet">
              <string>font-size: 16px; font-weight: bold;</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="currentCourseName">
             <property name="text">
              <string>暂无课程</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
             <property name="styleSheet">
              <string>font-size: 20px; color: #e74c3c;</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="currentCourseTeacher">
             <property name="text">
              <string>任课教师：暂无</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="currentCourseTime">
             <property name="text">
              <string>上课时间：--:-- 至 --:--</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="countdownLabel">
             <property name="text">
              <string>倒计时：00:00:00</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
             <property name="styleSheet">
              <string>font-size: 18px; font-weight: bold; color: #e74c3c;</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QFrame" name="nextCourseFrame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>下一节课</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
             <property name="styleSheet">
              <string>font-size: 16px; font-weight: bold;</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="nextCourseName">
             <property name="text">
              <string>暂无课程</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
             <property name="styleSheet">
              <string>font-size: 20px; color: #3498db;</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="nextCourseTeacher">
             <property name="text">
              <string>任课教师：暂无</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="nextCourseTime">
             <property name="text">
              <string>上课时间：--:-- 至 --:--</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_1">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <!-- 右侧：课表展示 -->
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>本周课表</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="styleSheet">
           <string>font-size: 18px; font-weight: bold;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTableView" name="classTableView">
          <property name="minimumSize">
           <size>
            <width>600</width>
            <height>0</height>
           </size>
          </property>
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectRows</enum>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <!-- 底部：滚动通知栏 -->
    <item>
     <widget class="QLabel" name="marqueeLabel">
      <property name="text">
       <string>欢迎使用教室班牌信息展示系统 - 暂无滚动通知</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="styleSheet">
       <string>background-color: #fff3cd; color: #856404; padding: 10px; border-radius: 5px;</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="styleSheet">
    <string>background-color: #f8f9fa;</string>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="10" margin="15"/>
 <resources/>
 <connections>
  <connection>
   <sender>classComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>onClassSelected(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>200</x>
     <y>100</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>searchEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>onSearchTextChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>400</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>100</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>exportBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onExportBtnClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>600</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>100</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>noticeManagerBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onNoticeManagerBtnClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>700</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>700</x>
     <y>100</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>settingsBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onSettingsBtnClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>800</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>800</x>
     <y>100</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onClassSelected(int)</slot>
  <slot>onSearchTextChanged(QString)</slot>
  <slot>onExportBtnClicked()</slot>
  <slot>onNoticeManagerBtnClicked()</slot>
  <slot>onSettingsBtnClicked()</slot>
 </slots>
</ui>